torch/testing/_internal/distributed/_tensor/common_dtensor.py

```diff
@@ -140,27 +139,31 @@ def __init__(self, dim, hidden_dim, dropout_p):
     def forward(self, x):
         return self.resid_dropout(self.w2(self.gelu(self.w1(x))))
 
+
 class TransformerBlock(nn.Module):
     def __init__(self, args: ModelArgs):
         super().__init__()
         self.attention_norm = nn.LayerNorm(args.dim)
         self.attention = Attention(args)
         self.ffn_norm = nn.LayerNorm(args.dim)
-        self.feed_forward = FeedForward(args.dim, hidden_dim=4 * args.dim, dropout_p=args.dropout_p)
+        self.feed_forward = FeedForward(
+            args.dim, hidden_dim=4 * args.dim, dropout_p=args.dropout_p
+        )
 
     def forward(self, x):
         h = x + self.attention(self.attention_norm(x))
         out = h + self.feed_forward(self.ffn_norm(h))
         return out
 
+
 # A toy transformer model, partly inspired by the nanoGPT model:
 # https://github.com/karpathy/nanoGPT.
 class Transformer(nn.Module):
     def __init__(self, args: ModelArgs):
         super().__init__()
         assert args.vocab_size is not None
         assert args.max_seq_len is not None
-        self.max_seq_len = args.max_seq_len
+        self.model_args = args
```

Comment:
I saved the `ModelArgs` as attribute to avoid having to pass in the `ModelArgs` to `parallelize()`. Hopefully this is okay.

Reply:
not sure if there's any implication when we turn on torch.compile on 2D e2e tests 🤔 

I guess if there's issue pop up in the future, we can always refactor to pass args to parallelize directly

Reply:
To be safe, let me leave `self.max_seq_len` then.